src/test/system_tests.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+//
+#include <test/util/setup_common.h>
+#include <util/system.h>
+#include <univalue.h>
+
+#ifdef HAVE_BOOST_PROCESS
+#include <boost/process.hpp>
+#endif // HAVE_BOOST_PROCESS
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(system_tests, BasicTestingSetup)
+
+// At least one test is required (in case HAVE_BOOST_PROCESS is not defined).
+// Workaround for https://github.com/bitcoin/bitcoin/issues/19128
+BOOST_AUTO_TEST_CASE(dummy)
+{
+    BOOST_CHECK(true);
+}
+
+#ifdef HAVE_BOOST_PROCESS
+
+bool checkMessage(const std::runtime_error& ex)
+{
+    // On Linux & Mac: "No such file or directory"
+    // On Windows: "The system cannot find the file specified."
+    const std::string what(ex.what());
+    BOOST_CHECK(what.find("file") != std::string::npos);
+    return true;
+}
+
+bool checkMessageFalse(const std::runtime_error& ex)
+{
+    BOOST_CHECK_EQUAL(ex.what(), std::string("RunCommandParseJSON error: process(false) returned 1: \n"));
+    return true;
+}
+
+bool checkMessageStdErr(const std::runtime_error& ex)
+{
+    const std::string what(ex.what());
+    BOOST_CHECK(what.find("RunCommandParseJSON error:") != std::string::npos);
+    return checkMessage(ex);
+}
+
+BOOST_AUTO_TEST_CASE(run_command)
+{
+    {
+        const UniValue result = RunCommandParseJSON("");
+        BOOST_CHECK(result.isNull());
+    }
+    {
+#ifdef WIN32
+        // Windows requires single quotes to prevent escaping double quotes from the JSON...
+        const UniValue result = RunCommandParseJSON("echo '{\"success\": true}'");
+#else
+        // ... but Linux and macOS echo a single quote if it's used
+        const UniValue result = RunCommandParseJSON("echo \"{\"success\": true}\"");
```

Comment:
In commit "[util] add RunCommandParseJSON" (31cf68a3ad1f0a5537c8419e2912b55fbfb88fa0)

This code seems nonsensical, and I spent like an hour trying to figure out how it might be working. I think what is happening is that boost's [build_arg](https://github.com/boostorg/process/blob/4fd88876013a9547b8bc7183d0096c86e9a1141f/include/boost/process/detail/posix/basic_cmd.hpp#L53) function is building this up into a vector of three strings:

- `echo`
- `"{"success":`
- `true}"`

and passing these as `argv` to [execve](https://linux.die.net/man/2/execve), which I guess gets the job done but is not the most straightforward thing. Might be better to pass an argument vector, or a normal bash command line instead of a fake command line string that uses boost's weird command line splitting code and escaping rules. This would just be for clarity though, since I think the current code probably should work.

On `#ifdef WIN32` code, I don't see how that code is working, and I wonder it is actually running and passing on windows with appveyor, or is just broken and should be removed. Since we don't seem to be enabling the shell (cmd.exe) as a boost process option, and since single quotes have no meaning on windows, and since windows doesn't have an echo.exe binary, it seems impossible that this could be working.

Reply:
"boost's weird command line splitting code and escaping rule" are what the user has to deal with when using `-signer=` in the followup PR, so it seems reasonable to have to test use that too.

I fixed a failed AppVeyor tests with this `#ifdef WIN32` line, so the code does run. The external signer Python tests also run on AppVeyor in the followup PR.  Maybe it's running in the power shell?